### PR TITLE
[TT-8845] Updating to Go 1.19

### PR DESCRIPTION
**Pull request by mativm02**
## Description

Updating go.mod and CI jobs to use go 1.19
## Related Issue

https://tyktech.atlassian.net/browse/TT-8845?atlOrigin=eyJpIjoiZWIzMjllZDAzZmIwNDZlMjkyM2UxODc5MjI4ZGNhZmIiLCJwIjoiaiJ9
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: repo-policy/main.tf
Comment (tbuchaillot) [inline repo-policy/main.tf:47]: @mativm02 changing the required tests in release-1.3 is not needed since we are not going to propagate this changes to release-1.3

**Review by tbuchaillot — CHANGES_REQUESTED** [review]

File: repo-policy/main.tf
Comment (mativm02) [reply repo-policy/main.tf]: thanks tom!

**Review by tbuchaillot — APPROVED** [review]

**Comment by sredxny** [comment]
@mativm02 what else do we need to merge this work?

**Comment by mativm02** [comment]
> @mativm02 what else do we need to merge this work?

@sredxny I'm waiting for devops team approval

**Comment by ermirizio** [comment]
@mativm02 All required changes in the code has been made, all required tests needs to pass before I apply the changes, so we can apply and merge in a synchronous way.

**Pull request merged**
